R/check-input-helpers.R

```diff
@@ -0,0 +1,390 @@
+  }
+  return(TRUE)
+}
+
+
+
+
+
+#' Check columns in data.frame don't have NA values
+#' @inheritParams check_columns_present
+#' @return Returns an string with a message if any of the column names
+#' have NA values, otherwise returns TRUE
+#'
+#' @keywords internal
+check_no_NA_present <- function(data, columns) {
+  for (x in columns){
+    if (anyNA(data[[x]])) {
+      msg <- paste0(
+        "Checking `data`: ",
+        sum(is.na(data[[x]])),
+        " values in column `",
+        x,
+        "`` are NA and corresponding rows will be removed. This is fine if not unexpected." # nolint
+      )
+      return(msg)
+    }
+  }
+  return(TRUE)
+}
+
+
+
+
+# print stuff
+diagnose <- function(data) {
+
+}
+
+#' Check that there are no duplicate forecasts
+#'
+#' @description
+#' Runs [get_duplicate_forecasts()] and returns a message if an issue is encountered
+#' @inheritParams get_duplicate_forecasts
+#' @return Returns an string with an error message if an issue is found,
+#' otherwise returns TRUE
+#'
+#' @keywords internal
+check_duplicates <- function(data, forecast_unit = NULL) {
+  check_duplicates <- get_duplicate_forecasts(data, forecast_unit = forecast_unit)
+
+  if (nrow(check_duplicates) > 0) {
+    msg <- paste0(
+      "There are instances with more than one forecast for the same target. ",
+      "This can't be right and needs to be resolved. Maybe you need to ",
+      "check the unit of a single forecast and add missing columns? Use ",
+      "the function get_duplicate_forecasts() to identify duplicate rows"
+    )
+    return(msg)
+  }
+  return(TRUE)
+}
+
+
+# Function to check input for methods
+# there should not be a name clash between a metric and a column name
+#    --> maybe this should be checked by the actual method that computes scores
+# check whether any column name is a scoringutils metric
+# clashing_colnames <- intersect(colnames(data), available_metrics())
+# if (length(clashing_colnames) > 0) {
+#   clashing_colnames <- paste0('"', clashing_colnames, '"')
+#   warnings <- c(
+#     warnings,
+#     paste0(
+#       "At least one column in the data ",
+#       "(", toString(clashing_colnames), ") ",
+#       "corresponds to the name of a metric that will be computed by ",
+#       "scoringutils. Please check `available_metrics()`"
+#     )
+#   )
+# }
+
+
+#' Check column names are present in a data.frame
+#' @param data A data.frame or similar to be checked
+#' @param columns names of columns to be checked
+#' @return Returns string with a message with the first issue encountered if
+#'  any of the column names are not in data, otherwise returns TRUE
+#'
+#' @keywords check-inputs
+check_columns_present <- function(data, columns) {
+  colnames <- colnames(data)
+  for (x in columns){
+    if (!(x %in% colnames)) {
+      msg <- paste0("Data needs to have a column called '", x, "'")
+      return(msg)
+    }
+  }
+  return(TRUE)
+}
+
+#' Test whether all column names are present in a data.frame
```

Comment:
what is the difference between a check and a test?

Reply:
check returns TRUE on success and an error message otherwise. test returns TRUE on success and FALSE otherwise, assert returns TRUE on success and errors otherwise. Noted the need for documentation in #357